externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/repl/KyuubiSparkILoop.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark.repl
+
+import java.io.{ByteArrayOutputStream, File}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interpreter.JPrintWriter
+
+import org.apache.spark.SparkContext
+import org.apache.spark.repl.{Main, SparkILoop}
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.util.MutableURLClassLoader
+
+private[spark] case class KyuubiSparkILoop private (
+    spark: SparkSession,
+    output: ByteArrayOutputStream)
```

Comment:
because the scala operation is always running with sync, so maybe we can save the KyuubiSparkILoop output to a file and read it likes operation log.

Reply:
Looks like a good idea.